### PR TITLE
Adds design document for DTLS 1.3 implementation documentation

### PR DESCRIPTION
**Pull request by fwh-dc**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by mattcaswell — APPROVED** [review]
Nice work!

**Comment by t8m** [comment]
The check_docs CI failure is relevant.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
@mattcaswell please reconfirm

**Comment by davidben** [comment]
A few TLSWG threads to call to your attention. As we've been reviewing the documents, we noticed a few hiccups in parts of the protocol:
https://mailarchive.ietf.org/arch/msg/tls/6y8wTv8Q_IPM-PCcbCAmDOYg6bM/
https://mailarchive.ietf.org/arch/msg/tls/3IEIzc2ssdkZbyYEXIvz6t3ZhbI/
https://mailarchive.ietf.org/arch/msg/tls/_ku3-YDcroNmG_QKZsYTtqYzC0M/
https://mailarchive.ietf.org/arch/msg/tls/GYX_teYy5CTFiGCBgbQJQwv_Fj4/

(Thoughts welcome on the TLSWG list.)

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
doc check CI is relevant

**Review by t8m — APPROVED** [review]

**Review by mattcaswell — CHANGES_REQUESTED** [review]
Looks good just one very minor typo

File: doc/designs/dtlsv1_3/dtlsv1_3-main.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+When the session keys are updated the DTLSv1.3 implementation will clear the
+messages waiting for retransmission that belongs to the key that is being updated.
+
+#### TLS 1.3 "compatibility mode" is not enabled
+
+Opposed to the TLS 1.3 implementation, the DTLSv1.3 implementation does not offer
+the compatibility which is in consistency with RFC9147 section 5.
+
+This enforced by the macro `SSL_CONNECTION_MIDDLEBOX_IS_ENABLED(sc)`.
```
[the remark this answers is not in the document]
Comment (fwh-dc) [reply doc/designs/dtlsv1_3/dtlsv1_3-main.md]: Fixed!

**Comment by fwh-dc** [comment]
ping @mattcaswell

**Review by mattcaswell — APPROVED** [review]

**Comment by t8m** [comment]
Squashed and merged to the feature branch. Thank you for your contribution.

**Pull request closed without merging**